### ISSUE TITLE
The default project icon is not Android adaptive icon-friendly

### ISSUE THREAD

**Issue by alexzheng** [godotengine/godot#38170]
**Godot version:**
3.2.1

**OS/device including version:**
MI Mix 2 
**Issue description:**
When using the default godot icon, the launcher icon is cropped when export using the 3.2.1.
In 3.1.2, all the icon is visible.  
**Steps to reproduce:**

**Minimal reproduction project:**
![Screenshot_2020-04-24-17-34-56-353_com miui home](https://user-images.githubusercontent.com/1130047/80198628-ee334880-8652-11ea-98bb-03325a2366b5.jpg)

**Comment by Calinou** [comment]
This is because Android adaptive icons are now used, but the default Godot icon isn't designed to be an adaptive icon.

**Comment by alexzheng** [comment]
If no adaptive icon is provided, is it possible to export the main icon just as 3.1 did?

**Comment by Calinou** [comment]
@alexzheng I believe this is already what Godot is doing. It's just that there's no graceful fallback provided. Maybe the icon could be scaled down and a black rectangle could be added behind it, but this would be a fair amount of work for little benefit.

**Comment by alexzheng** [comment]
I mean if no adaptive icon is provided, export launcher icon using the main icon just as 3.1 did, NO CROP.

**Comment by Calinou** [comment]
@alexzheng *Android* is cropping the icon here, not Godot. This is done so the user can choose between many icon styles (circle, rounded square, square, …).

**Comment by nathanfranke** [comment]
Related to #34868

Should we add an option to go back to the non-adaptive icon? (Or is that implied if Adaptive Foreground/Background are not defined)

**Comment by alexzheng** [comment]
Yes, this is what exactly I mean.
If no Adaptive icons is provided, just fall back to the original style.

**Comment by Calinou** [comment]
> If no Adaptive icons is provided, just fall back to the original style.

@MadEqua Do you think this can be done?

**Comment by MadEqua** [comment]
@alexzheng Can you provide a screenshot of the launcher icon section of the Android export window? Just to make sure I don't make any wrong assumptions about what's happening.

@Calinou That can be done by removing the adaptive icons from the export template on export time, forcing Android to use the older ones. However, removing stuff from the apk in runtime it's not as easy as just replacing (as it's being done now) because binary metadata needs to be updated and adjusted to still have a valid apk. IMO it's not worth all the work and extra complexity because ultimately every project will want to provide it's own well designed icons. A simpler option is to simplify the fallback chain to just use the default Godot adaptive icon when the user doesn't provide any, instead of falling back to the regular ones, which may become cropped, but this may be worse for some people.

**Comment by alexzheng** [comment]
@MadEqua  All the tree options are empty in the luncher icons section.
<img width="453" alt="Screen Shot 2020-05-06 at 7 54 45 AM" src="https://user-images.githubusercontent.com/1130047/81126784-fc486980-8f6e-11ea-9ad4-68439440f6a3.png">

**Comment by MadEqua** [comment]
Thank you @alexzheng. 
So yes, the foreground icon is falling back to the project icon, which is getting cropped on your Android launcher.

I guess a simple and reasonable solution is to simplify the fallback chain from:

> Provided foreground icon -> Provided main icon -> Project icon -> Default Godot foreground icon.

to

> Provided foreground icon -> Default Godot foreground icon.

The downside is that users will be forced to provide an adaptive icon if they don't want the default Godot one. If more people in the community think this is a good idea, I'll put up a PR.

**Comment by alexzheng** [comment]
No, fallback to either Provided main icon or Project icon  without cropping, just as the 3.1 did would be nice, but not the default godot icon.

I would  just provide a Project icon for all platform.

**Comment by alexzheng** [comment]
Maybe this title is misleading, I'm not care about how the default project icon is cropped, but all of my other custom icons are cropped, these icons show just OK when export by preview godot release.

**Comment by MadEqua** [comment]
> I would just provide a Project icon for all platform.

I guess it's possible for Godot to resize the adaptive icons fallbacks. It's not perfect but it may work for most cases. I'll test that.

> I'm not care about how the default project icon is cropped, but all of my other custom icons are cropped, 

I see that now. At the moment, your best solution is change your icons for to be adaptive as explained in the [docs](https://docs.godotengine.org/en/latest/getting_started/workflow/export/exporting_for_android.html#providing-launcher-icons), it's not that hard if you follow the Google design guidelines, and it will look great on Android 8+.  It's something that everyone that develops for Android has to deal with.

**Comment by akien-mga** [comment]
Why close this one? It's still valid and while it can be worked around, it's still a bad out-of-the-box experience.

**Comment by m4gr3d** [comment]
I closed it first because of the lack of activity on the issue (both from the op and @MadEqua).

The technical reason though is that it may be hard to strike the balance of providing the requested functionality while not further confusing users. As if we are able to provide the functionality (to be confirmed by @MadEqua), how do we help differentiate between where users should provide an adaptive icon and where they should provide a regular one, or do we forgo adaptive icons all together?

**Comment by EarthDLL** [comment]
[en]
**Any progress so far?**
Different projects have different requirements. I tried to delete the adaptive icon section in the template, but it seems to have no effect after exporting. And in the case of Gradle building, I don't have the code to modify this feature.
I think the best solution is to use only a single icon when there is no adaptive icon set.

[cn]
目前有任何进展？
不同项目需求不同。我尝试删除模板中的自适应图标部分，但导出后似乎没有作用。并且在gradle构建的情况下，我没有可以修改这个功能的代码。
我认为最好的解决方案就是在没有设置自适应图标时，仅使用单一图标
